src/RegistryCoordinator.sol

```diff
@@ -507,6 +509,12 @@ contract RegistryCoordinator is
         if (operatorId == 0) {
             operatorId = blsApkRegistry.registerBLSPublicKey(operator, params, pubkeyRegistrationMessageHash(operator));
         }
+        if (_operatorInfo[operator].status == OperatorStatus.DEREGISTERED && 
```

Comment:
probably should be an `else if` rather than just an `if`, even if the cases are (supposed to be) mutually exclusive?